src/librbd/io/Utils.cc

```diff
@@ -112,13 +112,26 @@ void read_parent(I *image_ctx, uint64_t object_no, const Extents &extents,
 
   ldout(cct, 20) << dendl;
 
+  if (extents->size() > 1) {
+    on_finish = new LambdaContext([on_finish, extents](int r) {
+        if (r == 0) {
+          auto data = extents->front().bl;
+          for (auto& extent: *extents) {
+            data.splice(0, extent.length, &extent.bl);
```

Comment:
Doesn't this corrupt the buffer when it splices into itself? Perhaps create a new `C_ReadFromParent` context that owns a `bufferlist` that can be passed to `aio_read` and whose finish callback splices the data back into the original buffers.